### PR TITLE
Remove fDebug3,4, and net and convert to BCLog::LogFlags

### PR DESCRIPTION
**Pull request by jamescowens**
This PR removes the fDebug3, 4 and net conditionals from all logs, converting to the category SCRAPER, NET, or RPC as appropriate. For the scraper, the use of the category is internal to the _log function, and is used to control the posting of INFO level log entries only. Critical, error, and warning level messages will still be logged in both logs whether the SCRAPER category is on or off.

Right now I think we have to hold off converting fDebug2, fDebug10 (the massively noisy one), and the baseline fDebug, because it will generate too many conflicts with the work cycy is doing.

So at this point, vanilla debug statements are still controlled by debug=true. You can set additional categories by putting debug=<category> in the config file. This setting may be repeated for as many categories as you need by default. A category can be set or unset with the logging rpc command.

Note that after @cyrossignol gets the other big PR merged into development, we will then collapse away the rest of the old fDebug flags. The debug directive will then represent turning on a preset set of categories if "true" is specified, or by category when the individual category is specified.

### CODE REVIEW

**Comment by jamescowens** [comment]
I am testing this now and will remove from draft once I am satisfied.

**Comment by jamescowens** [comment]
Tests good for me...

File: src/init.cpp [hunk 1]
```diff
@@ -605,15 +604,17 @@ bool AppInit2(ThreadHandlerPtr threads)
 
     fDebug=false;
 
-    if (fDebug)
-        fDebugNet = true;
-    else
-        fDebugNet = GetBoolArg("-debugnet");
-
```
Comment (jamescowens) [inline src/init.cpp:612]: This code never worked anyway! It is before the pull from GetArg... ! But I implemented the intent in the change...

File: src/init.cpp [hunk 2]
```diff
@@ -1164,7 +1151,7 @@ bool AppInit2(ThreadHandlerPtr threads)
 
     if (pindexBest->nVersion <= 10) {
         uiInterface.InitMessage(_("Loading Network Averages..."));
-        if (fDebug3) LogPrintf("Loading network averages");
+        LogPrint(BCLog::LogFlags::NET, "Loading network averages");
```
Comment (cyrossignol) [inline src/init.cpp]:
```suggestion
        LogPrint(BCLog::LogFlags::TALLY, "Loading network averages");
```

**Review by cyrossignol — COMMENTED** [review]
Looks good. Just one minor change.

File: src/init.cpp (hunk 1 above)
Comment (cyrossignol) [reply src/init.cpp:612]: `LogFlags::NET` is pretty chatty now since it inherited some entries from `debug10`. May want to drop this in the future.

Comment (jamescowens) [reply src/init.cpp:612]: Maybe we should just drop it now, since someone can always turn the category on if they want. I think I will do that.

Comment (jamescowens) [reply src/init.cpp:612]: Took it out.

**Comment by jamescowens** [comment]
Ready to go.

**Review by cyrossignol — APPROVED** [review]

**Pull request merged**